src/node/internal/internal_zlib.ts

```diff
@@ -6,10 +6,10 @@
 import { default as zlibUtil, type ZlibOptions } from 'node-internal:zlib';
 import { Buffer } from 'node-internal:internal_buffer';
 import { validateUint32 } from 'node-internal:validators';
-import { ERR_INVALID_ARG_TYPE } from 'node-internal:internal_errors';
-import { isArrayBufferView } from 'node-internal:internal_types';
 import { Zlib } from 'node-internal:internal_zlib_base';
 
+type CompressCallback = (error: Error | null, result: Buffer | null) => void;
```

Comment:
This type is wrong. Technically this supports (null, null), but it is not possible.

```suggestion
type CompressCallback = ((error: Error, result: null) => void) | ((null, result: Buffer) => void)
```

Reply:
Thanks, that's much better.

Reply:
Hmm, I'm not sure the new type is right either. To me it says "a function that handles errors" OR "a function that handles results". You can't pass in a function that handles both with that type. FWIW, DefinitelyTyped uses a type like the one I had originally.